HpToolsLauncher/Runners/AlmTestSetsRunner.cs

```diff
@@ -1217,10 +1222,9 @@ public override TestSuiteRunResults Run()
             //start test runner
             if (filteredTestList.Count == 0)
             {
-                //ConsoleWriter.WriteErrLine("Specified test not found on ALM, please check your test path.");
                 //this will make sure run will fail at the end. (since there was an error)
-                //Launcher.ExitCode = Launcher.ExitCodeEnum.Failed;
-                Console.WriteLine(Resources.AlmTestSetsRunnerNoTestAfterApplyingFilters);
+                ConsoleWriter.WriteErrLine(Resources.AlmTestSetsRunnerNoTestAfterApplyingFilters);
+                Launcher.ExitCode = Launcher.ExitCodeEnum.Failed;
```

Comment:
Do we really need to treat this scenario as a `Failed` state? 

Say, I'm a customer and I want to run all **unstable** ALM tests. So I do apply a status filter `unstable`. First time, I can get two unstable tests that will be run. And then I run the job again, this time since there is no unstable tests, no filtered tests are returned so eventually I received a **Failed** job result. It sounds not good for me. I may just want this job to run periodically to make sure there is **unstable** tests left, otherwise, everything is good.

Reply:
We should only return `Failed` state in this case, if no filter is selected using the `IsFilterSelected` flag. Through the `FilterTests` method not only filters are applied, but tests are searched by name, if a specific test is given by the user, not a test set.

Reply:
_Update_: will only return `Failed` state if a specific test is given by the user, otherwise won't.